### DISCUSSION TITLE
Load applications when opening the page Applications

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit#16127] [Ideas]
Hello Cockpit developers,

It took me a while before I realized that I had to click on the refresh button on the page Applications to load the applications (and see them). I think it's good idea to load them immediately after you open the page Applications. Then, I think, the button refresh isn't necessary anymore.

Or just load them the first time.

**Comment by garrett** [comment]
Thanks for reporting this!

The page should load the applications the first time without having to click refresh. The refresh button is only there to force a refresh when someone manually updates the appstream data.

If it's not loading for you the first time, this is definitely a bug and not intended behavior.

I'll transfer this to an issue. Could you answer the following please?

- Which Linux distribution (and version of the distribution) are you using?
- Which version of Cockpit are you on?
- Which browser (and version) are you using?

We did recently update the page, perhaps you might be hitting a corner case that our tests didn't catch. (It does load automatically here on Fedora 34 with the latest version of Cockpit build from the source repository, on Firefox 90.)

**Reply by garrett** [reply]
Oh, I can't transfer this, but I was able to open an issue from this discussion @ https://github.com/cockpit-project/cockpit/issues/16128. (It's also in the sidebar of this page, under "events".)

**Reply by Verhoeckx** [reply]
Great to know that this isn't intended behavior but probably just a bug!

Linux distribution: Red Hat Enterprise Linux 8.4
Version of Cockpit: 238.2 (release 1.el8)
Browser: Firefox 90.0

I will add the above information to the new issue!
